polaris-react/src/components/DataTable/DataTable.tsx

```diff
@@ -32,8 +32,8 @@ export interface DataTableProps {
   totals?: TableData[];
   /** Custom totals row heading */
   totalsName?: {
-    singular: string;
-    plural: string;
+    singular: string | React.ReactNode;
```

Comment:
You should be able to just have `React.ReactNode` since that type includes [many other types](https://github.com/facebook/react/blob/d5f1b067c8bbb826b823d0354a28ba31078b70c0/packages/shared/ReactTypes.js#L10-L16), including `string`